src/zenml/integrations/kserve/model_deployers/kserve_model_deployer.py

```diff
@@ -0,0 +1,464 @@
+        for item in response.get("items", []):
+            snake_case_item = self._camel_to_snake(item)
+            inference_service = V1beta1InferenceService(**snake_case_item)
+            inference_services.append(inference_service)
+        return inference_services
+
+    def _camel_to_snake(self, obj: Dict[str, Any]) -> Dict[str, Any]:
+        if isinstance(obj, (str, int, float)):
+            return obj
+        if isinstance(obj, dict):
+            assert obj is not None
+            new = obj.__class__()
+            for k, v in obj.items():
+                new[self._convert_to_snake(k)] = self._camel_to_snake(v)
+        elif isinstance(obj, (list, set, tuple)):
+            assert obj is not None
+            new = obj.__class__(self._camel_to_snake(v) for v in obj)
+        else:
+            return obj
+        return new
+
+    def _convert_to_snake(self, k: str) -> str:
+        return re.sub(r"(?<!^)(?=[A-Z])", "_", k).lower()
+
+    def find_model_server(
+        self,
+        running: bool = False,
+        service_uuid: Optional[UUID] = None,
+        pipeline_name: Optional[str] = None,
+        pipeline_run_id: Optional[str] = None,
+        pipeline_step_name: Optional[str] = None,
+        model_name: Optional[str] = None,
+        model_uri: Optional[str] = None,
+        model_type: Optional[str] = None,
+        predictor: Optional[str] = None,
+    ) -> List[BaseService]:
+        """Find one or more KServe model services that match the given criteria.
+
+        Args:
+            running: If true, only running services will be returned.
+            service_uuid: The UUID of the service that was originally used
+                to deploy the model.
+            pipeline_name: name of the pipeline that the deployed model was part
+                of.
+            pipeline_run_id: ID of the pipeline run which the deployed model was
+                part of.
+            pipeline_step_name: the name of the pipeline model deployment step
+                that deployed the model.
+            model_name: the name of the deployed model.
+            model_uri: URI of the deployed model.
+            model_type: the implementation specific type/format of the deployed
+                model.
+            predictor: the name of the predictor that was used to deploy the model.
+
+        Returns:
+            One or more Service objects representing model servers that match
+            the input search criteria.
+        """
+        config = KServeDeploymentConfig(
+            pipeline_name=pipeline_name or "",
+            pipeline_run_id=pipeline_run_id or "",
+            pipeline_step_name=pipeline_step_name or "",
+            model_uri=model_uri or "",
+            model_name=model_name or "",
+            predictor=predictor or "",
+            resources={},
+        )
+        labels = config.get_kubernetes_labels()
+
+        if service_uuid:
+            labels["zenml.service_uuid"] = str(service_uuid)
```

Comment:
Is this a string that is used somewhere else to set this label? If yes maybe we can make it a constant somewhere?